models/Element/Service.php

```diff
@@ -875,6 +876,13 @@ public static function renewReferences($data, $initial = true, $key = null)
             if ($data instanceof Model\AbstractModel) {
                 $properties = $data->getObjectVars();
                 foreach ($properties as $name => $value) {
+
+                    //do not renew object reference of ObjectAwareFieldInterface - as object might point to a
+                    //specific version of the object and must not be reloaded with DB version of object
+                    if(($data instanceof ObjectAwareFieldInterface || $data instanceof DataObject\Localizedfield) && $name === 'object') {
```

Comment:
@fashxp I think  `Localizedfield` should also implement `ObjectAwareFieldInterface`.  That should be safe as it's a `final` class, we can also change the signature of `setObject()` (remove `$markAsDirty` parameter).  `markAllLanguagesAsDirty()` is public and can be called separately if necessary. Additionally `__objectAwareFields` would need to be filled accordingly by the localized field. 
WDYT?

Reply:
Sorry, already covered in #13874 (PR) and #12598 (issue).